connector_jira/models/project_task/common.py

```diff
@@ -160,6 +165,50 @@ def name_get(self):
             names.append((task_id, name))
         return names
 
+    @api.model
+    def _get_connector_jira_fields(self):
+        return [
+            'jira_bind_ids',
+            'name',
+            'date_deadline',
+            'user_id',
+            'description',
+            'active',
+            'project_id',
+        ]
+
+    @api.model
+    def create(self, vals):
+        ProjectProject = self.env['project.project']
+        project_id = vals.get('project_id')
+        if project_id:
+            project_id = ProjectProject.sudo().browse(project_id)
+            if not self.env.context.get('connector_jira') and \
+                    project_id.mapped('jira_bind_ids')._is_linked():
+                raise exceptions.UserError(_(
+                    'Task can not be created in project linked to JIRA!'
+                ))
```

Comment:
I would extract the whole content in a method called by the `create` inherit, so one can override it if required.